Integrations/integration-Intezer.yml

```diff
@@ -0,0 +1,221 @@
+                case '403':
+                    throw '403 Forbidden - The API key is not valid\n';
+                case '404':
+                    throw '404 Not Found - Access to expired analysis URL or sha256 not found for analysis by sha256\n';
+                case '410':
+                    throw '410 Gone - Analysis no longer exists in the service\n';
+                case '413':
+                    throw '413 Payload Too Large - Large request\n';
+                case '415':
+                    throw '415 Unsupported Media Type - Wrong or missing content type\n';
+                case '500':
+                    throw '500 Internal Server Error - Internal error\n';
+                case '503':
+                    throw '503 Service Unavailable\n';
+                default:
+                    throw 'Failed to perform request ' + SERVER_API + url + ', request status code: ' + result.StatusCode + '\n';
+            }
+        }
+        if (result.Body === '' && result.StatusCode === 204) {
+            return { statusCode: result.StatusCode };
+        }
+        if (result.Body === '') {
+            throw 'No content received' + '\n';
+        }
+        var obj;
+        try {
+            obj = JSON.parse(result.Body);
+        } catch (ex) {
+            throw 'Error parsing reply - ' + result.Body + ' - ' + ex;
+        }
+        return { body: result.Body, obj: obj, statusCode: result.StatusCode };
+    };
+
+    var getAnalysis = function(analysisId) {
+        var analysisRes = sendRequest('POST', ANALYSIS_RESULT_URL + analysisId);
+        return analysisRes.obj;
+    };
+
+    var analyzeSHA256 = function(hash, codeItemType, maxRetries, delay) {
+        // validate file is indeed sha256
+        var entries = [];
+        if (!SHA_256_REGEXP.test(hash)) {
+            throw 'File hash is not valid sha256.\nIntezer file hash reputation supports only sha256 hash.\n\n';
+        }
+
+        // create file analysis request
+        var res = sendRequest('POST', HASH_URL, {
+            sha256: hash,
+            code_item_type: codeItemType
+        });
+
+        var analysisId;
+        if (res.statusCode === 201) {
+            analysisId = res.obj.analysis_id;
+        } else {
+            throw 'Failed to create sha256 analysis for ' + hash + ', request status code: ' + res.statusCode + '\n';
+        }
+
+        // perform wait loop
+        res = getAnalysis(analysisId);
+        var tries = 0;
+        while (res.status !== SUCCESS_STATUS && tries < maxRetries) {
+            // this means that status is 'in_progress' or 'queued'
+            tries++;
+            wait(delay); // wait {delay} seconds
+            res = getAnalysis(analysisId);
+        }
+
+        // if polling result did not work, return an error message
+        if (res.status !== SUCCESS_STATUS) {
+            throw 'Failed to pool sha256 analysis for ' + hash + ', Try to change maxRetries or delay arguments\n';
```

Comment:
not sure we want to fail in case we don't have .a response yet. @slavikm - wdyt? what did you do in other services? I remember you did some polling as well...

Reply:
@gshriki , this is after waiting for [delay * maxRetries].